bindings/bind_py_structure_manager.cc

```diff
@@ -616,6 +618,196 @@ namespace rascal {
         R"(Get the dense feature matrix associated with the calculator and
         the collection of structures (managers) using the list of keys
         provided. Only applicable when Calculator uses BlockSparseProperty.)");
+    manager_collection.def(
+        "get_features_gradient",
+        [](ManagerCollection_t & managers, const Calculator & calculator,
+           py::list & all_keys_l) {
+          using Manager_t = typename ManagerCollection_t::Manager_t;
+          using Prop_t =
+              typename Calculator::template PropertyGradient_t<Manager_t>;
+          using Keys_t = typename Prop_t::Keys_t;
+          if (managers.size() == 0) {
+            throw std::runtime_error(
+                R"(There are no structure to get features from)");
+          }
+          Keys_t all_keys;
+          if (all_keys_l.size() > 0) {
+            // convert the list of keys from python in to the proper type
+            for (py::handle key_l : all_keys_l) {
+              auto key = py::cast<std::vector<int>>(key_l);
+              all_keys.insert(key);
```

Comment:
what happen if an invalid key is passed by the user? I did not find a check that all keys are valid.

Reply:
what do you mean by invalid? there might be keys that are not in the representation at hand (and that's exactly the point of this possibility actually) so zeros will be inserted there.
If by invalid you mean that they can't be cast to std::vector<int> then an error will be raised by the relevant code.

Reply:
Ok, that's good!